service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/CaseExtensionController.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.service.ValidateGroupService;
+import uk.gov.hmcts.reform.fpl.validation.groups.CaseExtensionGroup;
+
+import java.time.LocalDate;
+
+import static org.springframework.util.StringUtils.isEmpty;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.DATE;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateToString;
+
+
+@Api
+@RestController
+@RequestMapping("/callback/case-extension")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CaseExtensionController {
+    private final ObjectMapper mapper;
+    private final ValidateGroupService validateGroupService;
+    private LocalDate caseCompletionDate;
+    private LocalDate eightWeekExtensionDate;
```

Comment:
I don't think it's a good idea to have private fields here, we should keep our controllers stateless and rather rely on case data.

Reply:
cool changed :)